l10n_ar_perceptions_basic/models/perception.py

```diff
@@ -60,7 +61,47 @@ class PerceptionPerception(models.Model):
         ],
         default='nacional',
     )
-    account_analytic_id = fields.Many2one(
-        'account.analytic.account',
-        string='Analytic Account',
-    )
+    active = fields.Boolean('Active', default=True, store=True, readonly=False)
+
+    @api.multi
+    def unlink(self):
+        field_obj = self.env['ir.model.fields']
+        fields = field_obj.search([
+            ('relation', '=', 'perception.perception')
+        ])
+        for perception in self:
+            for field in fields:
+                model_obj = self.env[field.model]
+                if field.ttype == 'one2many':
+                    if not perception[field.relation_field]:
+                        continue
+                elif field.ttype == 'many2many':
+                    if field.relation_table:
+                        cr = self.env.cr
+                        query = """
+                            SELECT * FROM
+                        """
+                        query += field.relation_table
```

Comment:
DO NOT ADD STRINGS. Python comes with a lot of builtin tools to perform string concatenation which are:

- more readable
- more efficient
- more pythonic
- easier to maintain in the long term